formats/notion/convert-to-md.ts

```diff
@@ -0,0 +1,419 @@
+			const childList: string[] = [];
+			for (let i = 0; i < children.length; i++) {
+				const itemContent = children.item(i)?.textContent;
+				if (!itemContent) continue;
+				childList.push(itemContent);
+			}
+			content = childList;
+			if (content.length === 0) return;
+			break;
+		case 'text':
+			content = body.textContent ?? '';
+			if (content.length === 0) return;
+			break;
+	}
+
+	return {
+		title,
+		content,
+	};
+};
+
+const getNotionLinks = (
+	body: HTMLElement,
+	{
+		attachmentPaths,
+	}: {
+		attachmentPaths: string[];
+	}
+) => {
+	const links: NotionLink[] = [];
+
+	body.querySelectorAll('a').forEach((a) => {
+		const decodedURI = stripParentDirectories(
+			decodeURI(a.getAttribute('href') ?? '')
+		);
+		const id = getNotionId(decodedURI);
+
+		const attachmentPath = attachmentPaths.find((filename) =>
+			filename.includes(decodedURI)
+		);
+		if (id && decodedURI.endsWith('.html')) {
+			links.push({ type: 'relation', a, id });
+		} else if (attachmentPath) {
+			links.push({
+				type: 'attachment',
+				a,
+				path: attachmentPath,
+			});
+		}
+	});
+
+	return links;
+};
+
+const fixDoubleBackslash = (markdownBody: string) => {
+	// Persistent error during conversion where backslashes in full-path links written as '\\|' become double-slashes \\| in the markdown.
+	// In tables, we have to use \| in internal links. This corrects the erroneous \\| in markdown.
+
+	const slashSearch = /\[\[[^\]]*(\\\\)\|[^\]]*\]\]/;
+	const doubleSlashes = markdownBody.match(new RegExp(slashSearch, 'g'));
+	doubleSlashes?.forEach((slash) => {
+		markdownBody = markdownBody.replace(
+			slash,
+			slash.replace(/\\\\\|/g, '\u005C|')
+		);
+	});
+
+	return markdownBody;
+};
+
+const formatDatabases = (body: HTMLElement) => {
+	// Notion includes user SVGs which aren't relevant to Markdown, so change them to pure text.
+	const users = body.querySelectorAll(
+		'span[class=user]'
+	) as NodeListOf<HTMLSpanElement>;
+	users.forEach((user) => {
+		user.innerText = user.textContent ?? '';
+	});
+
+	const checkboxes = body.querySelectorAll('td div[class*=checkbox]');
+	checkboxes.forEach((checkbox) => {
+		const newCheckbox = document.createElement('span');
+		newCheckbox.setText(
+			checkbox.className.contains('checkbox-on') ? 'X' : ''
+		);
+		checkbox.replaceWith(newCheckbox);
+	});
+
+	const selectedValues = body.querySelectorAll(
+		'table span[class*=selected-value]'
+	);
+	selectedValues.forEach((select) => {
+		const lastChild = select.parentElement?.lastElementChild;
+		if (lastChild === select) return;
+		select.setText(select.textContent + ', ');
+	});
+
+	const linkValues = body.querySelectorAll(
+		'a[href]'
+	) as NodeListOf<HTMLAnchorElement>;
+	linkValues.forEach((a) => {
+		// Any <a> with an email, phone number, or non-URL value registers as an internal link in Obsidian. [email@gmail.com](email@gmail.com) will create a new note when clicked. This strips these erroneous links.
+		if (a.href.startsWith('app://obsidian.md')) {
```

Comment:
Actually this doesn't work on mobile, because the page isn't hosted on `app://obsidian.md`. What you are looking for are ones where `.getAttribute('href')` (be aware it's different from `.href`) does not contain the protocol indicator character `:`.

Reply:
I'll change this to be more reliable—originally it was detecting any `href` from Notion that wasn't a valid `https://` URI. I can just detect that directly.